File: library/src/main/java/org/coner/crispy_fish/filetype/staging/SimpleStringStagingLineReader.java
package org.coner.crispy_fish.filetype.staging;

import org.coner.crispy_fish.datatype.underscore_pairs.*;

import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.BESTTIME;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.CAR;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.CC;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.CLASS;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.DRIVER;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.NUMBER;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.PAXED;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.PENALTY;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.POS;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.RUN;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.TM;
import static org.coner.crispy_fish.filetype.staging.StagingLineKeys.TOD;

public class SimpleStringStagingLineReader implements StagingLineReader<String> {

    protected final UnderscorePairReader<String> underscorePairReader;

    public SimpleStringStagingLineReader(UnderscorePairReader<String> underscorePairReader) {
        this.underscorePairReader = underscorePairReader;
    }

    public String getRunNumber(String line) {
        return underscorePairReader.get(line, RUN);
    }

    public String getRegisteredClass(String line) {
        return underscorePairReader.get(line, CLASS);
    }

    public String getRegisteredNumber(String line) {
        return underscorePairReader.get(line, NUMBER);
    }

    public String getRunRawTime(String line) {
        return underscorePairReader.get(line, TM);
    }

    public String getRunPenalty(String line) {
        return underscorePairReader.get(line, PENALTY);
    }

    public String getRegisteredDriverName(String line) {
        return underscorePairReader.get(line, DRIVER);
    }

    public String getRegisteredDriverCar(String line) {
        return underscorePairReader.get(line, CAR);
    }

    public String getRegisteredDriverCarColor(String line) {
        return underscorePairReader.get(line, CC);
    }

    public String getDriverPositionInClassForDay(String line) {
        return underscorePairReader.get(line, POS);
    }

    public String getDriverBestTimeOfDay(String line) {
        return underscorePairReader.get(line, BESTTIME);
    }

    public String getRunPaxTime(String line) {
        return underscorePairReader.get(line, PAXED);
    }

    public String getRunTimestamp(String line) {
        return underscorePairReader.get(line, TOD);
    }
}

File: library/src/main/kotlin/org/coner/crispy_fish/filetype/staging/SimpleStringStagingLineReader.kt
package org.coner.crispy_fish.filetype.staging

import org.coner.crispy_fish.datatype.underscore_pairs.UnderscorePairReader

class SimpleStringStagingLineReader(
        private val underscorePairReader: UnderscorePairReader<String>
) : StagingLineReader<String> {

    override fun getRunNumber(line: String) = underscorePairReader.get(line, StagingLineKeys.RUN)

    override fun getRegisteredDriverClass(line: String) = underscorePairReader.get(line, StagingLineKeys.CLASS)

    override fun getRegisteredDriverNumber(line: String) = underscorePairReader.get(line, StagingLineKeys.NUMBER)

    override fun getRunRawTime(line: String) = underscorePairReader.get(line, StagingLineKeys.TM)

    override fun getRunPenalty(line: String) = underscorePairReader.get(line, StagingLineKeys.PENALTY)

    override fun getRegisteredDriverName(line: String) = underscorePairReader.get(line, StagingLineKeys.DRIVER)

    override fun getRegisteredDriverCar(line: String) = underscorePairReader.get(line, StagingLineKeys.CAR)

    override fun getRegisteredDriverCarColor(line: String) = underscorePairReader.get(line, StagingLineKeys.CC)

    override fun getDriverPositionInClassForDay(line: String) = underscorePairReader.get(line, StagingLineKeys.POS)

    override fun getDriverBestTimeOfDay(line: String) = underscorePairReader.get(line, StagingLineKeys.BESTTIME)

    override fun getRunPaxTime(line: String) = underscorePairReader.get(line, StagingLineKeys.PAXED)

    override fun getRunTimestamp(line: String) = underscorePairReader.get(line, StagingLineKeys.TOD)
}
